src/BSON/UTCDateTime.c

```diff
@@ -275,6 +276,45 @@ PHP_METHOD(UTCDateTime, toDateTime)
 }
 /* }}} */
 
+/* {{{ proto array UTCDateTime::jsonSerialize()
+*/
+PHP_METHOD(UTCDateTime, jsonSerialize)
+{
+	php_phongo_utcdatetime_t *intern;
+	char                      s_milliseconds[24];
+	int                       s_milliseconds_len;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_UTCDATETIME_OBJ_P(getThis());
+
+	s_milliseconds_len = snprintf(s_milliseconds, sizeof(s_milliseconds), "%" PRId64, intern->milliseconds);
+
+	array_init_size(return_value, 1);
+
+#if PHP_VERSION_ID >= 70000
+	{
+		zval udt;
+
+		array_init_size(&udt, 1);
+		ADD_ASSOC_STRINGL(&udt, "$numberLong", s_milliseconds, s_milliseconds_len);
+		ADD_ASSOC_ZVAL_EX(return_value, "$date", &udt);
```

Comment:
I disagree that we need to change this now, as we had the exact same discussion in the HHVM PR, where we decided to do it that way.

The team meeting notes only say:

> Our JsonSerializable output should be in a format that libbson and read and convert back into BSON. This means that we may not be able to use formats in the Extended JSON spec just yet.